packages/nextjs/src/config/loaders/valueInjectionLoader.ts

```diff
@@ -0,0 +1,26 @@
+import { LoaderThis } from './types';
+
+type LoaderOptions = {
+  values: Record<string, unknown>;
+};
+
+/**
+ * Set values on the global/window object at the start of a module.
+ *
+ * Options:
+ *   - `values`: A record where the keys correspond to the keys of the global values to set and the values
+ *        correspond to the values of the values on the global object. Values must be JSON serializable.
```

Comment:
```suggestion
 *   - `values`: An object where the keys correspond to the keys of the global values to set and the values
 *        correspond to the values of the values on the global object. Values must be JSON serializable.
```

L: Since `Record` is really just a TS helper type (not an actual data type), I think it's clearer to call `values` an object.

M: Why must values be JSONifiable? Couldn't someone add something like `global.mySet = new Set([x, y, z])`?

Reply:
> M: Why must values be JSONifiable? Couldn't someone add something like global.mySet = new Set([x, y, z])?

It was just a restriction I put up just so we can keep the logic in the loader as simple as possible. Having the data be JSON serializable allows us to simply call `JSON.stringify` on it and it will always be valid JS and injectable into the module. One of us *could* try to inject a non-serializable value but they will soon realize that it won't work. In that case they should probably resort to the prefix loader anyways.